operator-common/src/main/java/io/strimzi/operator/common/metrics/MetricsHolder.java

```diff
@@ -161,6 +163,30 @@ public Counter lockedReconciliationsCounter(String namespace) {
                 "Number of reconciliations skipped because another reconciliation for the same resource was still running");
     }
 
+    /**
+     * Gauge metric for the max size recorded for the event queue.
+     *
+     * @param namespace Namespace of the resources being reconciled
+     *
+     * @return Metrics gauge
+     */
+    public AtomicInteger reconciliationsMaxQueueSize(String namespace) {
+        return getGauge(namespace, kind, METRICS_PREFIX + "reconciliations.max.queue.size",
+            metricsProvider, selectorLabels, reconciliationsMaxQueueMap, "Max size recorded for the shared event queue");
+    }
+
+    /**
+     * Gauge metric for the max size recorded for the event batch.
+     *
+     * @param namespace Namespace of the resources being reconciled
+     *
+     * @return Metrics gauge
+     */
+    public AtomicInteger reconciliationsMaxBatchSize(String namespace) {
```

Comment:
I don't especially like how the batch concept ends up leaking into the common `MetricsHolder` here. Non-batched operators are the norm. 

Another very minor gripe (which is out of scope for this PR) is that `MetricsHolder` (which is kind of a group of metrics) is bringing assumptions about namespaces. What that means for UTO is we end up having to pass around the `namespace` to a few classes which never cared about the namespace before, just so we have it to pass to the `MetricsHolder`. 

These points suggest that perhaps, eventually, we should think about refactoring the MetricsHolder to better cope be more flexible about the differences between different operators. In other words, perhaps there should be a `MetricsGroup` concept and a thing which returns a `MetricsGroup` for a given namespace, and different operators can have different metrics groups.

Reply:
I agree. For the moment, let me introduce a `BatchOperatorMetricsHolder` to host these new metrics. I'm also taking a note to implement the `MetricsGroup` concept as a follow up (along with few other notes I already have on this topic).